File: disastrousdoubling.py
#I attempted the same code in C++, however did not work (im assuming errors with long it) so I completed the code in python
#Time: 0.60 sec
tc, vals = input(), list(map(int, input().split()))
amnt = 1
for x in vals:
  amnt<<=1
  if x>amnt:
    amnt = -99
    break
  amnt-=x
print(amnt%1000000007 if amnt!=-99 else "error")

File: disastrousdoubling.rb
#Reshab Chhabra
#Time: 0.45s(2nd)
tc=gets
vals = gets.split(" ")
amnt = 1
vals.each do |x|
  x = x.to_i
  amnt<<=1
  if x > amnt
    amnt = -99
    break
  end
  amnt-=x
end
if amnt != -99
  puts(amnt%1000000007)
else
  puts('error')
end
